src/common/copilot/dataverseMetadata.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import fetch, { RequestInit } from "node-fetch";
+import path from "path";
+import * as vscode from "vscode";
+import fs from "fs";
+import yaml from 'yaml';
+
+interface Attribute {
+    LogicalName: string;
+}
+
+export async function getEntityColumns(entityName: string, orgUrl: string, apiToken: string): Promise<string[]> {
+    try {
+        const dataverseURL = `${orgUrl}api/data/v9.2/EntityDefinitions(LogicalName='${entityName}')?$expand=Attributes($select=LogicalName)`;
+        const requestInit: RequestInit = {
+            method: "GET",
+            headers: {
+                'Content-Type': "application/json",
+                Authorization: `Bearer ${apiToken}`,
+            },
+        };
+
+        const jsonResponse = await fetchJsonResponse(dataverseURL, requestInit);
+        const attributes = getAttributesFromResponse(jsonResponse);
+
+        return attributes.map((attribute: Attribute) => attribute.LogicalName);
+    } catch (error) {
+        return [];
+    }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+async function fetchJsonResponse(url: string, requestInit: RequestInit): Promise<any> {
+    const response = await fetch(url, requestInit);
+
+    if (!response.ok) {
+        throw new Error(`Network request failed with status ${response.status}`);
+    }
+
+    return response.json();
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function getAttributesFromResponse(jsonResponse: any): Attribute[] {
+    if (jsonResponse.Attributes && Array.isArray(jsonResponse.Attributes) && jsonResponse.Attributes.length > 0) {
+        return jsonResponse.Attributes;
+    }
+
+    return [];
+}
+
+
+export function getEntityName(): string {
+    let entityName = '';
+
+    try {
+        const activeEditor = vscode.window.activeTextEditor;
+
+        if (activeEditor) {
+            const document = activeEditor.document;
+            const absoluteFilePath = document.fileName;
+            const activeFileFolderPath = path.dirname(absoluteFilePath);
+            const activeFileName = path.basename(absoluteFilePath);
+            const fileNameFirstPart = activeFileName.split('.')[0];
```

Comment:
Will this work for advanced forms?

<img width="232" alt="image" src="https://github.com/microsoft/powerplatform-vscode/assets/908570/94334973-9685-49e3-b1a4-30c23b24aff5">

Reply:
Yes